tests/unit/test_configuration.py

```diff
@@ -555,18 +572,30 @@ def test_no_noise() -> None:
     # where all noise levels are 0.0
     no_noise_config = get_configuration("no_noise")
 
-    for dataset in no_noise_config.keys():
-        dataset_dict: LayeredConfigTree = no_noise_config[dataset]
-        dataset_row_noise_dict: LayeredConfigTree = dataset_dict[Keys.ROW_NOISE]
-        dataset_column_dict: LayeredConfigTree = dataset_dict[Keys.COLUMN_NOISE]
-        for row_noise_type in dataset_row_noise_dict.keys():
-            noise_specific_config: LayeredConfigTree = dataset_row_noise_dict[row_noise_type]
-            for key, value in noise_specific_config.items():
-                assert dataset_row_noise_dict[row_noise_type][key] == 0.0
-        for column in dataset_column_dict.keys():
-            column_noise_dict: LayeredConfigTree = dataset_column_dict[column]
-            for column_noise_type in column_noise_dict.keys():
-                assert column_noise_dict[column_noise_type][Keys.CELL_PROBABILITY] == 0.0
+    dataset_names = [dataset.name for dataset in DATASET_SCHEMAS]
+    for dataset in dataset_names:
+        dataset_schema = DATASET_SCHEMAS.get_dataset_schema(dataset)
+        for row_noise_type in dataset_schema.row_noise_types:
+            parameters = []
+            if row_noise_type.probability is not None:
+                parameters.append("row_probability")
+            if row_noise_type.additional_parameters is not None:
+                parameters.extend(list(row_noise_type.additional_parameters.keys()))
+            for parameter in parameters:
+                assert (
+                    no_noise_config.get_value(dataset, row_noise_type.name, parameter) == 0.0
+                )
+
+        dataset_columns = [column for column in dataset_schema.columns if column.noise_types]
+        for column in dataset_columns:
+            column_noise_types = [noise_type.name for noise_type in column.noise_types]
+            for column_noise_type in column_noise_types:
+                assert (
```

Comment:
I understand the test code wasn't checking the additional parameters of column noise types, but surely we should be doing that in the same way that we do it for row noise types?